### ISSUE TITLE
Connection reuse

### ISSUE THREAD

**Issue by mschirrmeister** [cloudflare/odoh-server-go#21]
Hello,

I noticed that the *odoh-server* creates a new TCP connection for each incoming query to the server(s) defined in the variable `nameServers`. Even if the connection from the previous query still exists.
Is it planned to implement some kind of connection reuse for performance reasons? Or is the idea to always run it next to a recursive resolver in the same local network, where the time for the tcp handshake can be ignored? But even in this situation, connection reuse could be a good thing.

**Comment by chris-wood** [comment]
> Is it planned to implement some kind of connection reuse for performance reasons? Or is the idea to always run it next to a recursive resolver in the same local network, where the time for the tcp handshake can be ignored? But even in this situation, connection reuse could be a good thing.

Yes, this is still planned! We would also welcome PRs, should you have free cycles to spare. :-)

**Comment by mschirrmeister** [comment]
I am unfortunately no developer :-(, otherwise I would definitely spare some time.
